core/shared/src/main/scala/zio/ZManaged.scala

```diff
@@ -39,7 +39,7 @@ final case class Reservation[-R, +E, +A](acquire: ZIO[R, E, A], release: Exit[An
  * has been consumed, the resource will not be valid anymore and may fail with
  * some checked error, as per the type of the functions provided by the resource.
  */
-final case class ZManaged[-R, +E, +A](reserve: ZIO[R, E, Reservation[R, E, A]]) { self =>
+final class ZManaged[-R, +E, +A] private (val reserve: ZIO[R, E, Reservation[R, E, A]]) extends Serializable { self =>
```

Comment:
Let's make `reserve` just a constructor parameter, and not a field:
```suggestion
final class ZManaged[-R, +E, +A] private (reservation: ZIO[R, E, Reservation[R, E, A]]) extends Serializable { self =>
```

And add an accessor method:
```scala
def reserve: ZIO[R, E, Reservation[R, E, A]] =  reservation
```